src/nlp/commands/dummy_data.py

```diff
@@ -0,0 +1,123 @@
+        self._dataset_name = path_to_dataset.split("/")[-2]
+
+    def run(self):
+        module_path = prepare_module(self._path_to_dataset)
+        builder_cls = import_main_class(module_path)
+
+        # use `None` as config if no configs
+        configs = builder_cls.BUILDER_CONFIGS or [None]
+
+        for config in configs:
+            if config is None:
+                version = builder_cls.VERSION
+            else:
+                version = config.version
+
+            dataset_builder = builder_cls(config=config)
+            mock_dl_manager = MockDownloadManager(
+                dataset_name=self._dataset_name, config=config, version=version, is_local=True
+            )
+
+            dummy_data_folder = os.path.join(self._path_to_dataset, mock_dl_manager.dummy_data_folder)
+            logger.info(f"Creating dummy folder structure for {dummy_data_folder}... ")
+            os.makedirs(dummy_data_folder, exist_ok=True)
+
+            try:
+                generator_splits = dataset_builder._split_generators(mock_dl_manager)
+            except FileNotFoundError as e:
+
+                print(
+                    f"Dataset {self._dataset_name} with config {config} seems to already open files in the method `_split_generators(...)`. You might consider to instead only open files in the method `_generate_examples(...)` instead. If this is not possible the dummy data has to be created with less guidance. Make sure you create the file {e.filename}."
+                )
+
+            files_to_create = []
+            split_names = []
+            dummy_file_name = mock_dl_manager.dummy_file_name
+
+            for split in generator_splits:
+                logger.info(f"Collecting dummy data file paths to create for {split.name}")
+                split_names.append(split.name)
+                gen_kwargs = split.gen_kwargs
+                generator = dataset_builder._generate_examples(**gen_kwargs)
+
+                try:
+                    dummy_data_guidance_print = "\n" + 30 * "=" + "DUMMY DATA INSTRUCTIONS" + 30 * "=" + "\n"
+                    config_string = f"config {config.name} of " if config is not None else ""
+                    dummy_data_guidance_print += (
+                        "- In order to create the dummy data for "
+                        + config_string
+                        + f"{self._dataset_name}, please go into the folder '{dummy_data_folder}' with `cd {dummy_data_folder}` . \n\n"
+                    )
+
+                    # trigger generate function
+                    for key, record in generator:
+                        pass
+
+                    dummy_data_guidance_print += f"- It appears that the function `_generate_examples(...)` expects one or more files in the folder {dummy_file_name} using the function `glob.glob(...)`. In this case, please refer to the `_generate_examples(...)` method to see under which filename the dummy data files should be created. \n\n"
```

Comment:
Does arriving at this point always mean that there is a `glob.glob` ?

Reply:
yeah actually I just checked the dataset scripts and this is quite an exception. I think there are only 2,3 datasets where this is releventa. One is `nlp/datasets/aeslc/aeslc.py`. 
I guess there could also be another command than `glob.glob()` that lists the files, but I think we always use `glob/glob` or @mariamabarham ?